Addition of a dataset

dilipkrishnamurthy opened the pull request:
Category: Materials Science
Dataset: Kamlet-Taft Parameters on a range of molecules
Data Type: {Pubchem CID, SMILES, alpha, beta}
Task Type: {Regression}
# Tasks: 2
# Compounds: 222
Rec - Split†: {Random}
Rec - Metric†: {MAE, RMSE} Reference: {Marcus, Yizhak. "The properties of organic liquids that are relevant to their use as solvating solvents." Chemical Society Reviews 22.6 (1993): 409-416.}

Comments to review:
yapf autoformatting carried out. Running into some issues while debugging (KeyError: 'ECFP' and HTTP import error from the AWS bucket); any help will be much appreciated. I will be happy to make the necessary changes and push again. 

kt_dataset.py: adheres to the template
kt_dateset_2.py: does not adhere to the template but was much easier to debug

Thanks!

ncfrey commented on deepchem/molnet/load_function/kt_dataset.py:
```diff
@@ -0,0 +1,239 @@
+"""
+Loading Kamlet-Taft (KT) parameter dataset 
+
+  KT parameters are scales to measure and quantify the Lewis acidity and basicity of molecules. The parameters are obtained through Nuclear Magnetic Resonance (NMR) spectra.
```
Can you change this to a one-line summary and move this information to the docstring of the load function?

ncfrey commented on deepchem/molnet/load_function/kt_dataset.py:
```diff
@@ -0,0 +1,239 @@
+  Set up kwargs to enable these transformations. Additional kwargs may
+  be given for specific featurizers, transformers, and splitters.
+
+  The load function must be modified with the appropriate DataLoaders
+  for all supported featurizers for your dataset.
+
+  Please refer to the MoleculeNet documentation for further information
+  https://deepchem.readthedocs.io/en/latest/moleculenet.html.
```
You can delete these notes and add relevant info for your dataset. In particular it would be helpful to include information about the size of the dataset, and any previous benchmarking studies that give an idea about what sort of performance should be expected.

ncfrey commented on deepchem/molnet/load_function/kt_dataset.py:
```diff
@@ -0,0 +1,239 @@
+
+  """
+
+  # # Warning message about this template
+  # raise ValueError("""
+  #   This is a template function and it doesn't do anything!
+  #   Use this function as a reference when implementing new
+  #   loaders for MoleculeNet datasets.
+  #   """)
```
This comment and some other notes about the template can be deleted.

ncfrey commented on deepchem/molnet/load_function/kt_dataset.py:
```diff
@@ -0,0 +1,239 @@
+      return kt_tasks, all_dataset, transformers
+
+  # First type of supported featurizers
+  supported_featurizers = []  # type: List[Featurizer]
```
Are there any supported featurizers to list here?

dilipkrishnamurthy commented:
Nathan,

Thanks for reviewing the pull request and for providing suggestions. I have incorporated the suggestions. I have also installed a local copy of DeepChem to which these files have been added and tested. I have a question about the unit test file that I have created:

- the sizes of train, test and valid are perfectly what's expected: 177, 22 and 23 = 222 (dataset size). However when I check the contents (alpha and beta) through, for e.g., train_dataset.y, things don't quite match up. If you take a look at the .csv, it will be obvious what I am referring to. 

I have also fixed some import issues that were present earlier. I look forward to your review.

Thanks,
Dilip

rbharath commented on deepchem/molnet/load_function/kt_dataset.py:
```diff
@@ -0,0 +1,210 @@
+
+  KT parameters are scales to measure and quantify the Lewis acidity and basicity of molecules. The parameters are obtained through Nuclear Magnetic Resonance (NMR) spectra. Random splitting is recommended for this dataset.
+  The raw data csv file contains columns below:
+  Please refer to https://arxiv.org/pdf/2008.08078 (2020) for which the dataset was hand curated. Benchmark training is also present in this work. Accuracy levels (rmse) reported here are of the order 0.01  for alpha and beta on unseen data. 
```
Could you add this to the references section? Here's an example (taken from the numpy docs guide https://numpydoc.readthedocs.io/en/latest/format.html)

```
Please refer to reference [1]_ 

References
-----------
.. [1] O. McNoleg, "The integration of GIS, remote sensing,
   expert systems and adaptive co-kriging for environmental habitat
   modelling of the Highland Haggis using object-oriented, fuzzy-logic
   and neural-network techniques," Computers & Geosciences, vol. 22,
   pp. 585-588, 1996.
```

rbharath replied:
Ah just saw you already have a references section. The arxiv paper can be added there :)

dilipkrishnamurthy replied:
Thanks a lot, Bharath. I have updated the file accordingly.

rbharath commented on deepchem/molnet/load_function/kt_dataset.py:
```diff
@@ -0,0 +1,210 @@
+  >> n_features = train_dataset.get_data_shape()[0]
+  >> model = dc.models.MultitaskClassifier(n_tasks, n_features)
+"""
+
```
Can you remove this whitespace?

dilipkrishnamurthy replied:
Thanks. I have also run yapf to check if formatting is fine. It loads as expected without errors on testing with my local copy.

rbharath commented on deepchem/molnet/load_function/kt_dataset.py:
```diff
@@ -0,0 +1,210 @@
+          os.path.join(data_dir, 'ktdataset.tar.gz'), data_dir)
+
+    # Changer loader to match featurizer and data file type
+    loader = deepchem.data.DataLoader(
```
This should be `CSVLoader` as well here

dilipkrishnamurthy replied:
Thanks. Good catch, I updated it.

rbharath commented on deepchem/molnet/load_function/kt_dataset.py:
```diff
@@ -0,0 +1,210 @@
+      deepchem.utils.download_url(url=KTDATASET_CSV_URL, dest_dir=data_dir)
+
+    loader = deepchem.data.CSVLoader(
+        tasks=kt_tasks, smiles_field="csmiles", featurizer=featurizer)
```
Could you use `feature_field` instead of `smiles_field`? 

```
loader = deepchem.data.CSVLoader(
    tasks=kt_tasks, feature_field="csmiles", featurizer=featurizer)
```

dilipkrishnamurthy replied:
Thanks. I made the change accordingly.

rbharath reviewed (commented):
I took a look through and the code looks pretty good on my end except for a few minor comments! I'm not sure what the unit test issue you're mentioning is though. Is the dataset not loading correctly?

Travis looks to be failing because of some formatting issues. Can you try running `yapf` locally and making sure it passes?
